test/gtest/ucp/ucp_test.cc

```diff
@@ -583,13 +577,57 @@ void ucp_test_base::entity::fence(int worker_index) const {
     ASSERT_UCS_OK(status);
 }
 
-void* ucp_test_base::entity::disconnect_nb(int worker_index, int ep_index,
-                                           enum ucp_ep_close_mode mode) const {
+void *ucp_test_base::entity::disconnect_nb(int worker_index, int ep_index,
+                                           enum ucp_ep_close_mode mode) {
     ucp_ep_h ep = revoke_ep(worker_index, ep_index);
     if (ep == NULL) {
         return NULL;
     }
-    return ucp_ep_close_nb(ep, mode);
+
+    void *req = ucp_ep_close_nb(ep, mode);
+    if (UCS_PTR_IS_PTR(req)) {
+        m_close_ep_reqs.push_back(req);
+        return req;
+    }
+
+    ASSERT_UCS_OK(UCS_PTR_STATUS(req));
+    return NULL;
+}
+
+bool ucp_test_base::entity::is_ep_closed(void *close_req) const {
+    return (close_req == NULL) ||
+           (ucp_request_check_status(close_req) != UCS_INPROGRESS);
+}
+
+void ucp_test_base::entity::close_ep_req_free(void *close_req) {
+    ASSERT_TRUE(is_ep_closed(close_req)) << "free not completed EP close request";
+    ASSERT_TRUE((close_req == NULL) ||
+                (ucp_request_check_status(close_req) == UCS_OK)) << "ucp_ep_close_nb failed";
+
+    if (close_req == NULL) {
+        return;
+    }
```

Comment:
maybe move this before line 604.. to simplify the assert

Reply:
done